hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/PrefixParser.java

```diff
@@ -0,0 +1,189 @@
+
+/**
+ * Tool that parses OM db file for prefix table.
+ */
+@CommandLine.Command(
+    name = "prefix",
+    description = "Parse prefix contents")
+@MetaInfServices(SubcommandWithParent.class)
+public class PrefixParser implements Callable<Void>, SubcommandWithParent {
+
+  @Spec
+  private CommandSpec spec;
+
+  @CommandLine.Option(names = {"--db"},
+      required = true,
+      description = "Database File Path")
+  private String dbPath;
+
+  @CommandLine.Option(names = {"--path"},
+      required = true,
+      description = "prefixFile Path")
+  private String filePath;
+
+  @CommandLine.Option(names = {"--bucket"},
+      required = true,
+      description = "bucket in /vol/bucket format")
+  private String bucket;
+
+  @CommandLine.Option(names = {"--volume"},
+      required = true,
+      description = "volume in /vol/bucket format")
+  private String volume;
+
+  public String getDbPath() {
+    return dbPath;
+  }
+
+  public void setDbPath(String dbPath) {
+    this.dbPath = dbPath;
+  }
+
+  @Override
+  public Class<?> getParentType() {
+    return OzoneDebug.class;
+  }
+
+  @Override
+  public Void call() throws Exception {
+    parse(volume, bucket, dbPath, filePath);
+    return null;
+  }
+
+  public static void main(String[] args) throws Exception {
+    new PrefixParser().call();
+  }
+
+  public void parse(String vol, String buck, String db,
+                    String file) throws Exception {
+    if (!Files.exists(Paths.get(db))) {
+      System.out.println("DB path not exist:" + db);
+      return;
+    }
+
+    System.out.println("FilePath is:" + file);
+    System.out.println("Db Path is:" + db);
+
+    OzoneConfiguration conf = new OzoneConfiguration();
+    conf.set(OMConfigKeys.OZONE_OM_DB_DIRS, db);
+    OzoneManagerRatisUtils.setBucketFSOptimized(true);
+
+    OmMetadataManagerImpl metadataManager =
+        new OmMetadataManagerImpl(conf);
+    metadataManager.start(conf);
+
+    org.apache.hadoop.fs.Path effectivePath =
+        new org.apache.hadoop.fs.Path("/");
+
+    Path p = Paths.get(file);
+
+    // First get the info about the bucket
+    String bucketKey = metadataManager.getBucketKey(vol, buck);
+    OmBucketInfo info = metadataManager.getBucketTable().get(bucketKey);
+    long lastObjectId = info.getObjectID();
+    WithParentObjectId objectBucketId = new WithParentObjectId();
+    objectBucketId.setObjectID(lastObjectId);
+    dumpInfo("Bucket", effectivePath, objectBucketId, bucketKey);
+
+    Iterator<Path> pathIterator =  p.iterator();
+    while(pathIterator.hasNext()) {
+      Path ss = pathIterator.next();
+      String path =
+          metadataManager.getOzonePathKey(lastObjectId, ss.toString());
+      OmDirectoryInfo directoryInfo =
+          metadataManager.getDirectoryTable().get(path);
+      effectivePath = getEffectivePath(effectivePath, ss.toString());
+
+      dumpInfo("Intermediate Dir", effectivePath,
+          directoryInfo, directoryInfo.getName());
```

Comment:
The file path to be parsed here is expected to be directory, right? If given path is a file, I don't think we will get the expected objID from dir table at the end of the while loop. If so, we should make sure input path is a directory.